### ISSUE TITLE
S3Client name already in use

### ISSUE THREAD

**Issue by ChrisTerBeke** [aws/aws-sdk-php#456]
Getting this error recently:

```
PHP Fatal error:  Cannot use Aws\S3Client as S3Client because the name is already in use in /base/data/home/apps/s~crack-talent-783/1.381614094212936608/vendor/aws/aws-sdk-php/src/S3/BucketStyleSubscriber.php on line 4
```

**Comment by ChrisTerBeke** [comment]
I use Laravel on Google App engine, never had this problem before until a few days ago.

**Comment by mtdowling** [comment]
Fixed by #458. We were trying to utilize a shallower namespace structure for clients and tried to change the namespaces in a way that allowed old code to still work via aliases at runtime. We ran into some edge cases (one you've run into here), and I've partially reverted those changes. This is now fixed in the v3 branch.
